src/main/java/com/google/sps/filters/SessionFilter.java

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.filters;
+ 
+import java.io.IOException;
+import java.util.logging.Logger;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import javax.servlet.http.Cookie;
+import javax.servlet.ServletContext;
+import java.util.stream.*;
+import java.util.Optional;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+/** Servlet filter that checks if the user has valid session to make a request. */
+public class SessionFilter implements Filter {
+
+    private ServletContext context;
+    private FilterConfig filterConfig;
+
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain filterChain) throws IOException, ServletException {
+
+        HttpServletRequest req = (HttpServletRequest) request;
+		HttpServletResponse resp = (HttpServletResponse) response;
+
+        if(userHasPermissions(req, resp)) {
+            filterChain.doFilter(request, response);
+        } else {
+            resp.sendRedirect("/homepage.html");
+            return;
+        }
+
+    }
+
+    /**
+    * Determines if the user has the right permissions to make the request based on their session id.
+    * @return boolean
+    */
+    public boolean userHasPermissions(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+
+        Cookie[] cookies = request.getCookies();
+
+        //user has not logged in, no cookies have been created
+        if(cookies == null) {
+            this.context.log("Unauthorized access request");
+            return false;
+        }
+
+        String sessionId = "";
+        
+        for(Cookie cookie : cookies) {
+            if(cookie.getName().equals("JSESSIONID")) {
```

Comment:
Is it really JSEESSIONID or should we name is SESSIONID?

Reply:
JSESSIONID is the default name the session gives to this cookie.